app/questionnaire/questionnaire_store_updater.py

```diff
@@ -321,6 +321,46 @@ def _capture_section_dependencies_for_answer(self, answer_id: str) -> None:
             else:
                 self.dependent_sections.add(DependentSection(section_id, None, None))
 
+    def _capture_section_dependencies_progress_value_source_for_current_section(
+        self,
+    ) -> None:
+        """
+        Captures a unique list of section ids that are dependents of the current section, for progress value sources.
+        """
+        current_section_id = self._current_location.section_id
+        progress_deps_mapping = (
+            self._schema.when_rules_section_dependencies_by_section_for_progress_value_source
+        )
+        for section_dependency_id in progress_deps_mapping:
+            if current_section_id in progress_deps_mapping[section_dependency_id]:
+                self.dependent_sections.add(
+                    DependentSection(section_dependency_id, None, None)
+                )
```

Comment:
Like the block dependencies below, we need to check if the section is repeating. Currently, if a section progress was used inside a repeating section, the unhappy path wouldn't work.